promql/promqltest/test.go

```diff
@@ -376,6 +413,22 @@ func (t *test) parseEval(lines []string, i int) (int, *evalCmd, error) {
 			break
 		}
 
+		if strings.Split(defLine, " ")[0] == "expect" {
+			annoType, expectedAnno, err := parseExpect(defLine, i)
+			if err != nil {
+				return i, nil, formatErr("%w", err)
+			}
+			if cmd.expectedCmds == nil {
+				cmd.expectedCmds = make(map[expectCmdType][]*expectCmd)
+			}
+			if _, ok := cmd.expectedCmds[annoType]; !ok {
+				cmd.expectedCmds[annoType] = make([]*expectCmd, 0, 1)
```

Comment:
Let's not use `make` for a tiny theoretical performance gain outside of a performance critical code path. `append` is designed to work with a `nil` slice as its first argument. You can just remove this whole `if` statement.